gnd/src/main/java/com/google/android/gnd/repository/FeatureRepository.java

```diff
@@ -114,15 +113,10 @@ private Completable updateLocalFeature(RemoteDataEvent<Feature> event) {
     return getFeature(featureMutation.getProjectId(), featureMutation.getFeatureId());
   }
 
-  // TODO: Don't require projectId to be the active project.
   @Cold
   public Single<Feature> getFeature(String projectId, String featureId) {
     return projectRepository
-        .getProjectLoadingState()
-        .compose(Loadable::values)
-        .firstElement()
-        .filter(project -> project.getId().equals(projectId))
-        .switchIfEmpty(Single.error(() -> new NotFoundException("Project not found: " + projectId)))
+        .getProject(projectId)
```

Comment:
What if the local db was deleted while the app was paused? In that case the project will not be in the local db and this will crash.

Reply:
It shouldn't crash if the downstream handles the exception and displays/logs it. But I'm not sure if that is currently being handled or not.

Reply:
Imo this shouldn't depend on the order in which the project and feature are loaded from remote, but even if we always try to sync from remote when this is called we'd need to handle the failure case when there's not network connection and the project isn't in the local db.

Instead, perhaps we can just document the assumption that this only works if the project and feature are already cached, and return a more informative error if either of them are not?